File: atividade_2/main/src/Paciente.java
public class Paciente {
    private long cpf;
    private String nome_completo;
    private String cartao_sus;
    private TIPO_ATENDIMENTO tipo_atendimento;

    private Paciente esquerda;
    private Paciente direita;

    public Paciente(long cpf, String nome_completo, String cartao_sus, TIPO_ATENDIMENTO tipo_atendimento) {
        this.cpf = cpf;
        this.nome_completo = nome_completo;
        this.cartao_sus = cartao_sus;
        this.tipo_atendimento = tipo_atendimento;
    }

    public long getCpf() {
        return cpf;
    }

    public void setCpf(long cpf) {
        this.cpf = cpf;
    }

    public String getNome_completo() {
        return nome_completo;
    }

    public void setNome_completo(String nome_completo) {
        this.nome_completo = nome_completo;
    }

    public String getCartao_sus() {
        return cartao_sus;
    }

    public void setCartao_sus(String cartao_sus) {
        this.cartao_sus = cartao_sus;
    }

    public TIPO_ATENDIMENTO getTipo_atendimento() {
        return tipo_atendimento;
    }

    public void setTipo_atendimento(TIPO_ATENDIMENTO tipo_atendimento) {
        this.tipo_atendimento = tipo_atendimento;
    }

    public Paciente getEsquerda() {
        return esquerda;
    }

    public void setEsquerda(Paciente esquerda) {
        this.esquerda = esquerda;
    }

    public Paciente getDireita() {
        return direita;
    }

    public void setDireita(Paciente direita) {
        this.direita = direita;
    }
}

File: atividade_3/src/Paciente.cpp
//
// Created by Lucas Perônico Barbotin on 21/09/26.
//

#include "Paciente.h"

#include <iostream>
#include <stdexcept>

Paciente::Paciente(long cpf, std::string nomeCompleto, std::string cartaoSus, TipoAtendimento tipoAtendimento) {
    setCpf(cpf);
    set_nome_completo(nomeCompleto);
    set_cartao_sus(cartaoSus);
    set_tipo_atendimento(tipoAtendimento);
    set_esquerda(nullptr);
    set_direita(nullptr);
}

void Paciente::printPaciente(Paciente* p) {
    std::cout << "CPF Do paciente: " + std::to_string(p->getCpf()) << std::endl;
    std::cout << "Nome do paciente: " + p->getNomeCompleto() << std::endl;
    std::cout << "Cartão do Sus: " + p->getCartaoSus() << std::endl;
    std::cout << "Tipo atendimento: "<< to_string(p->tipo_atendimento()) << std::endl;
    std::cout << std::endl;
}
